doc/examples/qsv_transcode.c

```diff
@@ -0,0 +1,442 @@
+                goto fail;
+            }
+            /* set AVCodecContext Parameters for encoder, here we keep them stay
+             * the same as decoder.
+             */
+            encoder_ctx->time_base = av_inv_q(decoder_ctx->framerate);
+            encoder_ctx->pix_fmt   = AV_PIX_FMT_QSV;
+            encoder_ctx->width     = decoder_ctx->width;
+            encoder_ctx->height    = decoder_ctx->height;
+            if ((ret = str_to_dict(optstr, &opts)) < 0) {
+                fprintf(stderr, "Failed to set encoding parameter.\n");
+                goto fail;
+            }
+            /* There is no "framerate" option in commom option list. Use "-r" to
+            * set framerate, which is compatible with ffmpeg commandline. The
+            * video is assumed to be average frame rate, so set time_base to
+            * 1/framerate. */
+            e = av_dict_get(opts, "r", NULL, 0);
+            if (e) {
+                encoder_ctx->framerate = av_d2q(atof(e->value), INT_MAX);
+                encoder_ctx->time_base = av_inv_q(encoder_ctx->framerate);
+            }
+            if ((ret = avcodec_open2(encoder_ctx, enc_codec, &opts)) < 0) {
+                fprintf(stderr, "Failed to open encode codec. Error code: %s\n",
+                        av_err2str(ret));
+                goto fail;
+            }
+            av_dict_free(&opts);
+
+            if (!(ost = avformat_new_stream(ofmt_ctx, enc_codec))) {
+                fprintf(stderr, "Failed to allocate stream for output format.\n");
+                ret = AVERROR(ENOMEM);
+                goto fail;
+            }
+
+            ost->time_base = encoder_ctx->time_base;
+            ret = avcodec_parameters_from_context(ost->codecpar, encoder_ctx);
+            if (ret < 0) {
+                fprintf(stderr, "Failed to copy the stream parameters. "
+                        "Error code: %s\n", av_err2str(ret));
+                goto fail;
+            }
+
+            /* write the stream header */
+            if ((ret = avformat_write_header(ofmt_ctx, NULL)) < 0) {
+                fprintf(stderr, "Error while writing stream header. "
+                        "Error code: %s\n", av_err2str(ret));
+                goto fail;
+            }
+
+            initialized = 1;
+        }
+        frame->pts = av_rescale_q(frame->pts, decoder_ctx->pkt_timebase,
+                                  encoder_ctx->time_base);
+        if ((ret = encode_write(pkt, frame)) < 0)
+            fprintf(stderr, "Error during encoding and writing.\n");
+
+fail:
+        av_frame_free(&frame);
+        if (ret < 0)
+            return ret;
+    }
+    return 0;
+}
+
+int main(int argc, char **argv)
+{
+    const AVCodec *enc_codec;
+    int ret = 0;
+    AVPacket *dec_pkt;
+
+    if (argc < 5 || (argc - 5) % 2) {
+        av_log(NULL, AV_LOG_ERROR, "Usage: %s <input file> <encoder> <output file> \"<encoding option>\""
+               "{<frame-number> \"<encoding options>\"}\n", argv[0]);
```

Comment:
Usually using [] for optional arguments and <> for required arguments.

Reply:
I change the output to
`./qsv_transcode <input file> <encoder> <output file> "<encoding option>" {[frame-number "encoding options"]}`
I am not sure if this is correct.

Reply:
how about to change the text to 
`./qsv_transcode <input file> <encoder> <output file> ["encoding option"] [<frame-number> ["encoding options"]]`

Reply:
`[<frame-number> <"encoding option set 1">]` this group can be multiple. How can I show this information to user?

Reply:
I change the log this:
`qsv_transcode <input file> <encoder> <output file> ["encoding option set 0"] [<frame_number> <"encoding options set 1">]...`